src/Microsoft.Data/PrimitiveColumn.Sort.cs

```diff
@@ -0,0 +1,61 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.Data
+{
+    public partial class PrimitiveColumn<T> : BaseColumn
+        where T : unmanaged
+    {
+        public override BaseColumn Sort(bool ascending = true)
+        {
+            PrimitiveColumn<long> sortIndices = GetAscendingSortIndices() as PrimitiveColumn<long>;
+            return _Clone(sortIndices, !ascending);
+        }
+
+        internal override BaseColumn GetAscendingSortIndices()
+        {
+            // Is Comparer<T>.Default guaranteed to sort in ascending order?
```

Comment:
@eerhardt

Reply:
This is guaranteed for ALL `IComparer` implementations.

See https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.icomparer-1.compare?view=netframework-4.8

> Returns: A signed integer that indicates the relative values of x and y, as shown in the following table.

Value | Meaning
-- | --
Less than zero | x is less than y.
Zero | x equals y.
Greater than zero | x is greater than y.

Reply:
You can remove this comment (in the next PR if you want)

---
In reply to: [284780397](https://github.com/dotnet/corefxlab/pull/2660#discussion_r284780397) [](ancestors = 284780397)